generate.rb

```diff
@@ -0,0 +1,53 @@
+#!/usr/bin/env ruby
+
+require 'bundler'
+require 'net/http'
+
+lockfile = ARGV.first
+if lockfile.nil?
+  warn 'Usage: generate.rb /path/to/Gemfile.lock'
+  exit 1
+end
+
+unless File.exist? lockfile
+  warn "Lockfile #{lockfile} does not exist"
+  exit 1
+end
+
+unless File.readable? lockfile
+  warn "Lockfile #{lockfile} could not be read"
+  exit 1
+end
+
+# Parse the Gemfile.lock and build a map or gem name to version
+bundle = Bundler::LockfileParser.new(Bundler.read_file(lockfile))
+gem_name_to_version = bundle.specs.each_with_object({}) { |gem_object, acc| acc[gem_object.name] = gem_object.version }
+
+# Print a warning if the version defined in the component defintion is 
+# Older than the version in the Gemfile.lock
+def check_update(component_def, spec, gem_name_to_version)
+  component_def.each_line do |line|
+    # TODO: Some of the component files handle multiple versions with a case statement (for example net-ssh). 
```

Comment:
Current output
```
cas@cas-ThinkPad-T460p:~/working_dir/puppet-runtime$ ruby generate.rb ~/working_dir/bolt/Gemfile.lock
Update needed for: deep_merge 
Upgrade from 1.0.1 to 1.2.1

Update Needed for deep_merge:
!mirrorsource https://rubygems.org/downloads/deep_merge-1.2.1.gem

Update needed for: fast_gettext 
Upgrade from 0 to 1.1.2

Update needed for: ffi 
Upgrade from 1.9.25 to 1.11.1

Update Needed for ffi:
!mirrorsource https://rubygems.org/downloads/ffi-1.11.1.gem

Update needed for: gettext 
Upgrade from 3.2.2 to 3.2.9

Update needed for: minitar 
Upgrade from 0 to 0.8

Update needed for: net-ssh 
Upgrade from 0 to 5.2.0

Update needed for: puppet-resource_api 
Upgrade from 1.8.1 to 1.8.3

Update Needed for puppet-resource_api:
!mirrorsource https://rubygems.org/downloads/puppet-resource_api-1.8.3.gem

Update needed for: semantic_puppet 
Upgrade from 0 to 1.0.2
```
When a gem version is reported to be `0`, that indicates that multiple gems are handled in that definition, for now those need to be looked at by hand. The fast_gettext, gettext and deep_merge gems are shared by other projects so I will need to coordinate with them and figure out a path to upgrade. We are holding off on upgrading ffi, and puppet-resource-api for now. See the spreadsheet linked in PR comment for more details.